### PR TITLE
DOC-5399: Added CPK Gradle plugin content to the docs repo

### PR DESCRIPTION
**Pull request by AnnaK-R3**

### CODE REVIEW

**Review by yarivus — DISMISSED** [review]
LGTM

File: content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+```
+
+The "companion's" Maven coordinates will be:
+```
+ ${group}.${artifact}:${artifact}.corda.cpk:${version}
+```
+
+* `cordaEmbedded`: This configuration behaves similarly to `cordaProvided` in the sense that it declares a `compileOnly`
+dependency that is excluded from both the CPK contents and the published POM. The difference is that the dependent JAR
+is also added to a `META-INF/lib` folder inside the CorDapp's JAR and appended to the JAR's `Bundle-Classpath` manifest
```
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md:190]:
```suggestion
is also added to a `META-INF/privatelib` folder inside the CorDapp's JAR and appended to the JAR's `Bundle-Classpath` manifest
```

**Review by josephzunigadaly — APPROVED** [review]

**Review by yarivus — APPROVED** [review]
LGTM

File: content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+Applying this plugin to a project declares that the project should create a CorDapp in CPK format. The CPK format
+CorDapp is a ZIP file with a `.jar` extension, which is the output of the JAR task, along with the dependent JARs of that JAR.
+The plugin will not include any of Corda's own JARs among these dependencies, nor any JARs that should be provided by
+Corda, such as Kotlin or Quasar. The JAR should also contain sufficient OSGi metadata to be a valid OSGi bundle.
+
+## Applying the Plugin
+
+1. Depending on the number of modules of your Gradle project, choose one of the following options:
+
+* If your Gradle project contains multiple modules, perform the following actions:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md:24]: Can't see a preview but it looks like there should be indentation here for the second-level bullet points to work?

File: content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+* If your Gradle project contains a single module, apply both plugins together:
+
+```
+plugins {
+    id 'net.corda.cordapp.cordapp-configuration'
+    id 'net.corda.plugins.cordapp-cpk2'
+}
+```
+
+2. Declare both plugins' versions in `settings.gradle`:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md:49]:
```suggestion
2. Declare the versions of both plugins in `settings.gradle`:
```

File: content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+            "verifySameSignerAsMe": true
+        }
+    ]
+}
+```
+
+The `cordapp` dependencies are transitive, meaning that if CorDapp B declares a `cordapp` dependency on CorDapp A,
+and then CorDapp C declares a cordapp dependency on CorDapp B, CorDapp C will acquire compile-time dependencies on the
+JARs of both CorDapps A and B. Additionally, the `cordaProvided` dependencies of both A and B will be added to
+CorDapp C's `cordaProvided` configuration. This mechanism, is accomplished by
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md:173]:
```suggestion
CorDapp C's `cordaProvided` configuration. This mechanism is accomplished by
```

File: content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+publishing each CPK with a "companion" POM that contains the additional dependency information. The `cordapp-cpk2`
+plugin resolves these "companion" POMs transparently to the user, ensuring that CorDapps have the expected transitive relationships.
+
+In order for everything to work as intended, the "companion" POM must be published into the same repository as its
+associated JAR artifact. For a JAR with Maven coordinates:
+```
+ ${group}:${artifact}:${version}
+```
+
+The "companion's" Maven coordinates will be:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md:183]:
```suggestion
...then the "companion's" Maven coordinates will be:
```

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

File: content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md (hunk 2 above)
Comment (AnnaK-R3) [reply content/en/platform/corda/5.0/developing-applications/packaging/cpk-plugin.md]: Corrected the indentation.

**Review by paulmoloneyr3 — APPROVED** [review]

**Review by yarivus — COMMENTED** [review]
thx @AnnaK-R3

**Pull request merged**